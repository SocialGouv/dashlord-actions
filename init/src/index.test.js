const fs = require("fs");
const path = require("path");
const core = require("@actions/core");

const sampleConfig = jest
  .requireActual("fs")
  .readFileSync(path.join(__dirname, "..", "dashlord.yml"))
  .toString();

jest.mock("fs");

const { getOutputs, getSiteTools } = require("./index");

let inputs = {};

describe("should parse dashlord config", () => {
  beforeAll(() => {
    // Mock getInput
    jest.spyOn(core, "getInput").mockImplementation((name) => {
      return inputs[name];
    });
  });
  beforeEach(() => {
    // Reset inputs
    inputs = {};
  });
  test("when no input", async () => {
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const outputs = getOutputs();
    expect(outputs.sites).toMatchSnapshot();
  });

  test("when single invalid url input", async () => {
    inputs.url = "zfzef";
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const outputs = getOutputs();
    expect(outputs.sites).toMatchSnapshot();
  });

  test("when single valid url input", async () => {
    inputs.url = "https://www.free.fr";
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const outputs = getOutputs();
    expect(outputs.sites).toMatchSnapshot();
  });

  test("when multiple urls input", async () => {
    inputs.url = "https://www.free.fr,pouet,http://chez.com";
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const outputs = getOutputs();
    expect(outputs.sites).toMatchSnapshot();
  });

  test("and getSiteTools https://chez.com match", async () => {
    inputs.url = "https://chez.com";
    fs.existsSync.mockReturnValue(true);
    fs.readFileSync.mockReturnValue(sampleConfig);
    const tools = getSiteTools({
      url: "https://chez.com",
      repositories: ["chez/chez-ui", "chez/chez-api"],
      tools: {
        screenshot: false,
        updownio: false,
        stats: false,
        budget_page: false,
        testssl: true,
      },
    });
    core.info(`tools=${JSON.stringify(tools)}`);
    expect(tools).toEqual({
      screenshot: false,
      nmap: true,
      zap: true,
      wappalyzer: true,
      http: true,
      testssl: true,
      lighthouse: true,
      thirdparties: true,
      nuclei: false,
      updownio: false,
      dependabot: true,
      codescan: true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      stats: false,
      budget_page: false,
      404: true,
    });
  });
});
