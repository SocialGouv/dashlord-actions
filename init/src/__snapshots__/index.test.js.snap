// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse dashlord config when multiple urls input 1`] = `
[
  {
    "repositories": [
      "iliad/free-ui",
      "iliad/free-api",
    ],
    "subpages": [
      "https://www.free.fr",
    ],
    "title": "Free",
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://www.free.fr",
  },
  {
    "pages": [
      "/login",
      "/profile",
    ],
    "repositories": [
      "chez/chez-ui",
      "chez/chez-api",
    ],
    "subpages": [
      "http://chez.com",
      "http://chez.com/login",
      "http://chez.com/profile",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": false,
      "stats": false,
      "testssl": true,
      "thirdparties": true,
      "updownio": false,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "http://chez.com",
  },
  {
    "subpages": [
      "https://voila.fr",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://voila.fr",
  },
]
`;

exports[`should parse dashlord config when no input 1`] = `
[
  {
    "repositories": [
      "iliad/free-ui",
      "iliad/free-api",
    ],
    "subpages": [
      "https://www.free.fr",
    ],
    "title": "Free",
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://www.free.fr",
  },
  {
    "pages": [
      "/login",
      "/profile",
    ],
    "repositories": [
      "chez/chez-ui",
      "chez/chez-api",
    ],
    "subpages": [
      "http://chez.com",
      "http://chez.com/login",
      "http://chez.com/profile",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": false,
      "stats": false,
      "testssl": true,
      "thirdparties": true,
      "updownio": false,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "http://chez.com",
  },
  {
    "subpages": [
      "https://voila.fr",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://voila.fr",
  },
]
`;

exports[`should parse dashlord config when single invalid url input 1`] = `
[
  {
    "repositories": [
      "iliad/free-ui",
      "iliad/free-api",
    ],
    "subpages": [
      "https://www.free.fr",
    ],
    "title": "Free",
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://www.free.fr",
  },
  {
    "pages": [
      "/login",
      "/profile",
    ],
    "repositories": [
      "chez/chez-ui",
      "chez/chez-api",
    ],
    "subpages": [
      "http://chez.com",
      "http://chez.com/login",
      "http://chez.com/profile",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": false,
      "stats": false,
      "testssl": true,
      "thirdparties": true,
      "updownio": false,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "http://chez.com",
  },
  {
    "subpages": [
      "https://voila.fr",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://voila.fr",
  },
]
`;

exports[`should parse dashlord config when single valid url input 1`] = `
[
  {
    "repositories": [
      "iliad/free-ui",
      "iliad/free-api",
    ],
    "subpages": [
      "https://www.free.fr",
    ],
    "title": "Free",
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://www.free.fr",
  },
  {
    "pages": [
      "/login",
      "/profile",
    ],
    "repositories": [
      "chez/chez-ui",
      "chez/chez-api",
    ],
    "subpages": [
      "http://chez.com",
      "http://chez.com/login",
      "http://chez.com/profile",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": false,
      "stats": false,
      "testssl": true,
      "thirdparties": true,
      "updownio": false,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "http://chez.com",
  },
  {
    "subpages": [
      "https://voila.fr",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://voila.fr",
  },
]
`;

exports[`should parse dashlord config with inputs.tool 1`] = `
[
  {
    "repositories": [
      "iliad/free-ui",
      "iliad/free-api",
    ],
    "subpages": [
      "https://www.free.fr",
    ],
    "title": "Free",
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://www.free.fr",
  },
  {
    "pages": [
      "/login",
      "/profile",
    ],
    "repositories": [
      "chez/chez-ui",
      "chez/chez-api",
    ],
    "subpages": [
      "http://chez.com",
      "http://chez.com/login",
      "http://chez.com/profile",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": false,
      "stats": false,
      "testssl": true,
      "thirdparties": true,
      "updownio": false,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "http://chez.com",
  },
  {
    "subpages": [
      "https://voila.fr",
    ],
    "tools": {
      "404": true,
      "budget_page": true,
      "codescan": true,
      "declaration-a11y": true,
      "declaration-rgpd": true,
      "dependabot": true,
      "dsfr": true,
      "ecoindex": true,
      "http": true,
      "lighthouse": true,
      "nmap": true,
      "nuclei": false,
      "screenshot": true,
      "stats": true,
      "testssl": false,
      "thirdparties": true,
      "updownio": true,
      "wappalyzer": true,
      "zap": true,
    },
    "url": "https://voila.fr",
  },
]
`;
